env0/resource_environment.go

```diff
@@ -849,11 +861,9 @@ func getDeployPayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 		payload.BlueprintRevision = revision.(string)
 	}
 
-	if configuration, ok := d.GetOk("configuration"); ok {
+	if configuration, ok := d.GetOk("configuration"); ok && isRedeploy {
```

Comment:
Playing with the UI... it's only applicable for a redeploy. In the past, it was different?! Maybe?

Reply:
we can have configurations for initial deploy, doesn't that change break things in that scenario?

Reply:
The "update" operation calls the function "deploy" which calls "getDeployPayload" with isRedploy = true.
The "create" operation calls the function "getCreatePayload" which calls "getDeployPayload" with isRedploy = false.

In initial deploy "create" use-case, configuration is passed during the create of the environment, not during the first deploy (as I observed when using the UI + API).